roslibrust_codegen/src/gen.rs

```diff
@@ -257,10 +268,19 @@ fn parse_ros_value(ros_type: &str, value: &str, array_info: Option<Option<usize>
                 let vec_str = format!("{parsed:?}.iter().map(|x| x.to_string()).collect()");
                 quote! { #vec_str }
             } else {
-                // Halfass attempt to deal with ROS's string escaping / quote bullshit
-                let value = &value.replace('\'', "\"");
-                let parsed: String = serde_json::from_str(value).unwrap_or_else(|_| panic!("Failed to parse a literal value in a message file to the corresponding rust type: {value} to String"));
-                quote! { #parsed }
+                match version {
+                    RosVersion::ROS1 => {
+                        // For ROS1 then entire contents except for leading and trailing whitespace are used
+                        let value = value.trim();
+                        quote! { #value }
+                    }
+                    RosVersion::ROS2 => {
+                        // For ROS2 value must be in quotes, and either single or double quotes are okay
```

Comment:
Can a string be passed like `'this is a "string"'`

Reply:
Yeah we're almost certainly handling this wrong, but the ROS2 documentation is so vague about what correct behavior is I don't even know what to do:
https://docs.ros.org/en/rolling/Concepts/Basic/About-Interfaces.html?highlight=interface%20files#constants

![image](https://github.com/Carter12s/roslibrust/assets/8097919/2a115bbd-a5de-4071-8c94-6ac09b993274)

The documentation literally doesn't say anything about string values for ROS2 being in quotes, and we have to assume that I guess from the default values section, but it says default values aren't escaped, but nothing about constant values. Again I guess we assume they are not. I think I'll take another swing at some more robust logic and probably add a set of unit tests around it, just to document what we think valid is.

I think the only way to figure out what is "actually correct here" is to look at what the ros2 tools do, but that seems like work...

Reply:
I'm not sure what the appropriate channel is to ask for clarification, but our best path would be to get that section of the documentation clarified (even if the result is that lack of definition means it could be anything).